util/eventbus/subscribe.go

```diff
@@ -111,6 +106,21 @@ func (q *subscribeState) pump(ctx context.Context) {
 	}
 }
 
+func (s *subscribeState) snapshotQueue() []DeliveredEvent {
+	resp := make(chan []DeliveredEvent)
+	select {
+	case s.snapshot <- resp:
+		select {
+		case snapshot := <-resp:
+			return snapshot
+		case <-s.dispatcher.Done():
```

Comment:
I believe we have the same arrangement here as in the publisher.

Reply:
Yup, fixed.